src/TokenAirdrop.hs

```diff
@@ -43,10 +45,28 @@ tokenAirdrop config = do
         utxos <- utxosAt config $ config.ownAddress
         let lookups = Constraints.unspentOutputs utxos
 
-        submitTx @Void config pubKeyAddressMap lookups tx
+        eTxId <- submitTx @Void config pubKeyAddressMap lookups tx
+        case eTxId of
+          Left err -> pure $ Left err
+          Right txId -> do
+            putStrLn $ "Submitted transaction successfully: " ++ show txId
+            waitUntilHasTxIn config txId
+            pure $ Right ()
     )
     $ zipWith combine2To3 txPairs [1 :: Int ..]
 
+-- | Repeatedly waits a block until we have the inputs we need
```

Comment:
Perhaps the function should report an error after N blocks with no transaction?

Reply:
Its hard to define this - in the last 2 ish months we went from 1-2 blocks to near 45. hard coding this seems like an easy way to break the script following future congestion (though this should hopefully clear up in the coming months)

Reply:
I do wonder if theres some way to know if a transaction got rejected

Reply:
I've added a 50 block warning, but I don't want to make it error